client/src/app/pages/search/components/SearchMenu.tsx

```diff
@@ -0,0 +1,314 @@
+  }, [advisories, packages, sboms, vulnerabilities]);
+  return {
+    list: entityList,
+    defaultValue:
+      sbomFilterState.filterValues[FILTER_TEXT_CATEGORY_KEY]?.[0] || "",
+  };
+}
+
+export interface ISearchMenu {
+  onChangeSearch: (searchValue: string | undefined) => void;
+}
+
+export const SearchMenu: React.FC<ISearchMenu> = ({ onChangeSearch }) => {
+  const { list: entityList, defaultValue } = useAllEntities();
+
+  const [searchValue, setSearchValue] = React.useState<string | undefined>(
+    defaultValue
+  );
+  const [autocompleteOptions, setAutocompleteOptions] = React.useState<
+    React.JSX.Element[]
+  >([]);
+  const [isAutocompleteOpen, setIsAutocompleteOpen] =
+    React.useState<boolean>(false);
+
+  const searchInputRef: React.RefObject<HTMLInputElement> = React.useRef(null);
+  const autocompleteRef: React.RefObject<HTMLInputElement> = React.useRef(null);
+
+  const onChangeSearchValue = (newValue: string) => {
+    if (
+      newValue !== "" &&
+      searchInputRef &&
+      searchInputRef.current &&
+      searchInputRef.current.contains(document.activeElement)
+    ) {
+      setIsAutocompleteOpen(true);
+
+      // When the value of the search input changes, build a list of no more than 10 autocomplete options.
+      // Options which start with the search input value are listed first, followed by options which contain
+      // the search input value.
+      let options: React.JSX.Element[] = entityList
+        .filter(
+          (option) =>
+            option.id.toLowerCase().startsWith(newValue.toLowerCase()) ||
+            option.title?.toLowerCase().startsWith(newValue.toLowerCase()) ||
+            option.description?.toLowerCase().startsWith(newValue.toLowerCase())
+        )
+        .map((option) => (
+          <MenuItem
+            itemId={option.id}
+            key={option.id}
+            description={option.description}
+            to={option.navLink}
+          >
+            {option.title} <Label color={option.typeColor}>{option.type}</Label>
+          </MenuItem>
+        ));
+
+      if (options.length > 10) {
+        options = options.slice(0, 10);
+      } else {
+        options = [
+          ...options,
+          ...entityList
+            .filter(
+              (option: IEntity) =>
+                !option.id.startsWith(newValue.toLowerCase()) &&
+                option.id.includes(newValue.toLowerCase())
+            )
+            .map((option: IEntity) => (
+              <MenuItem itemId={option.id} key={option.id}>
+                {option.id}
+              </MenuItem>
+            )),
+        ].slice(0, 10);
+      }
+
+      // The menu is hidden if there are no options
+      setIsAutocompleteOpen(options.length > 0);
+      setAutocompleteOptions(options);
+    } else {
+      setIsAutocompleteOpen(false);
+    }
+
+    setSearchValue(newValue);
+  };
+
+  const onClearSearchValue = () => {
+    setSearchValue("");
+    onChangeSearch("");
```

Comment:
Perhaps `onChangeSearch("")` should not be called here? I tried to test it and when I start typing text and then click on the "X" icon that appears on the right of the input text then a filter "''" (empty string) is applied. But we should wait for the user to press down "Enter" or click on the "Search" button before applying a filter.